make distclean now fails with this error

rm -f TAGS ID GTAGS GRTAGS GSYMS GPATH tags
rm -rf ./.deps
rm -f Makefile
make[1]: Leaving directory '/root/frr/vtysh'
Making distclean in ospfclient
make[1]: Entering directory '/root/frr/ospfclient'
Makefile:562: ../ospfd/.deps/ospf_api.Plo: No such file or directory
Makefile:563: ../ospfd/.deps/ospf_dump_api.Plo: No such file or directory
make[1]: *** No rule to make target '../ospfd/.deps/ospf_dump_api.Plo'.  Stop.
make[1]: Leaving directory '/root/frr/ospfclient'
Makefile:469: recipe for target 'distclean-recursive' failed
make: *** [distclean-recursive] Error 1

--- a/ospfclient/Makefile.am
+++ b/ospfclient/Makefile.am
@@ -11,7 +11,7 @@ libfrrospfapiclient_la_LIBADD = ../lib/libfrr.la
 sbin_PROGRAMS = ospfclient
 
 libfrrospfapiclient_la_SOURCES = \
-	ospf_apiclient.c ../ospfd/ospf_dump_api.c ../ospfd/ospf_api.c
+	ospf_apiclient.c
 
 ospfapiheaderdir = $(pkgincludedir)/ospfapi
 

--- a/ospfclient/ospf_apiclient.c
+++ b/ospfclient/ospf_apiclient.c
@@ -49,6 +49,10 @@
 
 #include "ospf_apiclient.h"
 
+/* *sigh* ... can't find a better way to hammer this into automake */
+#include "ospfd/ospf_dump_api.c"
+#include "ospfd/ospf_api.c"
+
 DEFINE_MGROUP(OSPFCLIENT, "libospfapiclient")
 DEFINE_MTYPE_STATIC(OSPFCLIENT, OSPF_APICLIENT, "OSPF-API client")
 

